### ISSUE TITLE
test run segfaults (Intel 2018)

### ISSUE THREAD

**Issue by sbecuwe** [libAtoms/QUIP#106]
Hello

I compiled QUIP with GAP using Intel 2018 compilers using the Makefile.linux_x86_64_ifort_icc arch file. The aim is to use the QUIP library in LAMMPS, but I first wanted to get the standalone example working. Apart from the GAP related questions and "Would you like to compile with preconditioned minimisation support?", I always accepted the suggested answer in ```make config```.

I tried three options for enabling MKL:

```
DEFAULT_MATH_LINKOPTS =  -L${MKLROOT}/lib/intel64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
DEFAULT_MATH_LINKOPTS =  -mkl
DEFAULT_MATH_LINKOPTS =  -mkl=sequential
```

Compiling went fine, but running the example results in a segfault. (Using the library in LAMMPS also resulted in a crash.)

```
$ bash ./test_quip.sh
test_quip.sh: forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
quip               0000000000A4514D  Unknown               Unknown  Unknown
libpthread-2.17.s  00002B8DB048B5E0  Unknown               Unknown  Unknown
quip               00000000007A5749  dictionary_module        1861  Dictionary.f95
quip               0000000000967A90  c_dictionary_add_         362  libAtoms_utils_no_module.f95
quip               000000000096BD36  read_xyz                  850  xyz.c
quip               00000000008A393A  cinoutput_module_         565  CInOutput.f95
quip               00000000008A3CF3  cinoutput_module_         912  CInOutput.f95
quip               0000000000406D4A  MAIN__                    332  quip.f95
quip               0000000000AA7B82  Unknown               Unknown  Unknown
libc-2.17.so       00002B8DB0BBFC05  __libc_start_main     Unknown  Unknown
quip               0000000000404569  Unknown               Unknown  Unknown
```

Has anyone experience with / advice for compiling a serial version of QUIP using Intel 2018 compilers and MKL? Thanks!

Stefan

**Comment by gabor1** [comment]
When the makefile asks you about “GAP support” did you say yes? and did you download the GAP.tar.gz file and untarred it in the QUIP/src directory? 

You can also check out our docker at dockerhub.com, it’s called libatomsquip, that has everything precompiled (including lammps)

— Gábor

Pembroke College supports CARA. A Lifeline to Academics at Risk. http://www.cara.ngo/

**Comment by sbecuwe** [comment]
Yes, I did say yes and put the untarred GAP in the QUIP/src directory. I also tested without GAP, but the same error occurs. So I decided to test a few other compilers:

Test #2: gcc/gfortran 7.3.0 and OpenBLAS 0.2.20: test_quip.sh returns OK
Test #3: Intel 2017 compilers + MKL: test_quip.sh returns OK!

Also integration with LAMMPS seems to be ok now. So maybe a warning could be added about Intel 2018 compilers?

Regards
Stefan
